modules/local/prepare_prediction_input/templates/prepare_prediction_input.py

```diff
@@ -0,0 +1,223 @@
+        elif len(tool_allele_input) < len(allele_ls):
+            logging.warning(f"Provided alleles {allele_ls}, but only {tool_allele_input} are supported by {tool}. Continuing with supported alleles..")
+
+        return tool_allele_input
+
+    def has_valid_aas(peptide: str) -> bool:
+        """
+        Check if a peptide contains only valid amino acids.
+        """
+        valid_aas = "ACDEFGHIKLMNPQRSTVWY"
+        return all(aa in valid_aas for aa in peptide)
+
+
+def main():
+    args = Arguments()
+
+    # Parse alleles to uniform format
+    alleles_normalized = Utils.parse_alleles(args.alleles)
+    supported_alleles_dict = json.load(open(args.supported_alleles_json))
+    # Filter alleles based on supported alleles for each tool and write to json
+    tools_allele_input = {}
+    for tool in args.tools:
+        tool_allele_input = Utils.keep_supported_alleles(alleles_normalized, tool, supported_alleles_dict[tool])
+        tools_allele_input[tool] = ';'.join(tool_allele_input)
+    with open(f"{args.prefix}_allele_input.json", "w") as f:
+        json.dump(tools_allele_input, f)
+
+
+    # Parse input file to desired format of tools
+    df_input = pd.read_csv(args.input, sep="\t")
+    logging.info(f"Read file with {len(df_input)} peptides.")
+    # Filter peptides with invalid amino acids
+    df_input = df_input[df_input[args.peptide_col_name].apply(Utils.has_valid_aas)]
+    # Filter peptides based on user-defined length
+    if args.mhc_class == "I":
+        df = df_input[df_input[args.peptide_col_name].str.len().between(args.min_peptide_length_classI, args.max_peptide_length_classI)]
+    else:
+        df = df_input[df_input[args.peptide_col_name].str.len().between(args.min_peptide_length_classII, args.max_peptide_length_classII)]
+    if len(df) == 0:
+        raise ValueError("No peptides left after applying length filters! Aborting..")
+    logging.info(f"Filtered peptides based on length. {len(df)} peptides left for prediction..")
+
+
+    # Filter peptides based on tool length boundaries and adjust input format
+    if "mhcflurry" in args.tools and args.mhc_class == "I":
+        df_mhcflurry = df[df[args.peptide_col_name].str.len().between(MinLength.MHCFLURRY.value, MaxLength.MHCFLURRY.value)]
+        logging.info(f"Input for MHCflurry detected. Preparing {len(df_mhcflurry)} peptides for prediction..")
+        # Get every combination of sequence and allele and write them to csv with columns sequence and allele
+        df_mhcflurry['allele'] = [tools_allele_input['mhcflurry'].split(';')] * len(df_mhcflurry)
+        df_mhcflurry = df_mhcflurry.explode('allele').reset_index(drop=True)
+        df_mhcflurry.rename(columns={args.peptide_col_name: "peptide"}, inplace=True)
+        df_mhcflurry[['peptide','allele']].to_csv(f'{args.prefix}_mhcflurry_input.csv', index=False)
+
+    if "mhcnuggets" in args.tools and args.mhc_class == "I":
```

Comment:
Optimize code

Reply:
I would like to see less copy pasing of the if-else statement. The contant is the tool, I am sure that you can write this in a cleaner way 🤣